File: Java/2021-10-08/Encontro20/src/main/java/DAO/CategoriaDao.java
package DAO;

import FakeDB.*;
import Pojo.*;
import java.util.*;

public class CategoriaDao {
    
    public CategoriaDao(){
        EstoqueFakeDB.getCategorias();
   
    }
    
    // CRUD
    
    public void create(CategoriaPojo pojo){
        EstoqueFakeDB.getCategorias().add(pojo);

    }
    
    public CategoriaPojo read(int categoriaID){
        return this.search(categoriaID);
        
    }
    
    public ArrayList<CategoriaPojo> readAll(){
        
        return EstoqueFakeDB.getCategorias();
        
    }
    
    public void update(CategoriaPojo pojo){
        CategoriaPojo busca =  this.search(pojo.getCategoriaID());
        EstoqueFakeDB.getCategorias().remove(busca);
        EstoqueFakeDB.getCategorias().add(pojo);
        
        
    }
    
    public void delete(int categoriaID){
        CategoriaPojo busca = this.search(categoriaID);
        EstoqueFakeDB.getCategorias().remove(busca);
        
    }   
    
    private CategoriaPojo search(int id){
        
            return EstoqueFakeDB.getCategorias().stream()
                .filter(c -> c.getCategoriaID() == id)
                .findFirst()
                .orElse(null);
        
    
    }
}


/*
EstoqueFakeDB.getCategorias().stream()
                .filter(c -> c.getCategoriaID() == categoriaID)
                .findFirst()
                .orElse(null);

*/

File: 2021-11-04/TreinoCSharp/ConsoleApp/Dao/CategoriaDao.cs
using System;
using ConsoleApp.Service;
using ConsoleApp.Dao;
using ConsoleApp.Poco;
using ConsoleApp.FakeDB;


namespace ConsoleApp.Dao
{
    public class CategoriaDao
    {

       int c = EstoqueFakeDB.getCategorias();
   
    }
    
    // CRUD
    
    public void create(CategoriaPoco poco){
        EstoqueFakeDB.getCategorias().add(poco);

    }
    
    public CategoriaPoco read(int categoriaID){
        return this.search(categoriaID);
        
    }
    
    public List<CategoriaPoco> readAll(){
        
        return EstoqueFakeDB.getCategorias();
        
    }
    
    public void update(CategoriaPoco poco){
        CategoriaPoco busca =  this.search(poco.getCategoriaID());
        EstoqueFakeDB.getCategorias().remove(busca);
        EstoqueFakeDB.getCategorias().add(poco);
        
        
    }
    
    public void delete(int categoriaID){
        CategoriaPoco busca = this.search(categoriaID);
        EstoqueFakeDB.getCategorias().remove(busca);
        
    }   
    
    private CategoriaPoco busca(int id){
        
            return EstoqueFakeDB.getCategorias().stream()
                .filter(c -> c.getCategoriaID() == id)
                .findFirst()
                .orElse(null);
        
    
    }


    }

}




